src/hooks/useSarcophagusNegotiation.ts

```diff
@@ -133,24 +143,23 @@ export function useSarcophagusNegotiation() {
                 dispatch(
                   setArchaeologistException(arch.profile.peerId, {
                     code: ArchaeologistExceptionCode.STREAM_EXCEPTION,
-                    message,
+                    message
                   })
                 );
+                throw Error('stream exception');
               })
               .finally(() => stream.close());
           } catch (e) {
-            // TODO: `message` will (likely) be user-facing. Need friendlier verbiage.
-            const message = `Exception occurred attempting to open stream to: ${arch.profile.archAddress}`;
-            console.error(`Stream exception on ${arch.profile.peerId}`, e);
-            dispatch(
-              setArchaeologistException(arch.profile.peerId, {
-                code: ArchaeologistExceptionCode.CONNECTION_EXCEPTION,
-                message,
-              })
-            );
+            throw Error(`stream exception: ${e}`);
           }
         })
-      );
+      ).catch((error) => {
+        throw Error(`Error retrieving arch signatures ${error}`);
+      });
```

Comment:
Will throwing at this point not cause `161` below to be skipped? Or do we actually want to throw here? 
I dunno, all that nesting we've gotten ourselves into is kinda confusing to parse by eye

Reply:
Yeah you are most likely correct -- this entire routine needs to be refactored, so this can get handled in a separate PR, but thanks for pointing it out.